### PR TITLE
Some BMP consistency + endpoints.json endpoint to list all available endpoints and parameters

### PR DESCRIPTION
**Pull request by AndyTempel**

### CODE REVIEW

File: server.py [hunk 1]
```diff
@@ -78,6 +78,11 @@ def index():
     return render_template('index.html', data=data)
 
 
+@app.route('/endpoints.json', methods=['GET'])
+def endpoints():
+    return jsonify({"endpoints": [{'name': x, 'parameters': y.params} for x, y in sorted(endpoints.items())]})
```
Comment (devoxin) [inline server.py]: Does this *need* to be sorted?

Comment (GeoffreyWesthoff) [reply server.py]: Not necessarily, but it would be nice perhaps

Comment (AndyTempel) [reply server.py]: alphabetical order or bean! Well if you think it's useless I can remove it in a jiffy.

File: endpoints/tweet.py [hunk 2]
```diff
@@ -39,7 +39,7 @@ def generate(self, avatars, text, usernames):
         tag_raw = usernames[1] if len(usernames) == 2 else usernames[0]
         text3 = wrap(font3, f'@{tag_raw}', 1150)
 
-        time = datetime.now().strftime('%-I:%M %p - %d %b %Y')
```
Comment (GeoffreyWesthoff) [inline endpoints/tweet.py]: Krom and I already talked about this. Keep the '%-I', it makes it 12H which is more consistent with Twitter

**Review by GeoffreyWesthoff — CHANGES_REQUESTED** [review]

File: server.py (hunk 1 above)
Comment (devoxin) [reply server.py]: Tbf if the primary use-case is to dynamically support endpoints within bots then A-Z sorting isn't a huge priority here and just adds a few ms of overhead at most

Comment (GeoffreyWesthoff) [reply server.py]: I agree.

**Review by GeoffreyWesthoff — APPROVED** [review]

File: endpoints/tweet.py (hunk 2 above)
Comment (AndyTempel) [reply endpoints/tweet.py]: didn't i revert this? Back to your version

**Review by devoxin — APPROVED** [review]

**Pull request merged**